### PR TITLE
feat(self-hosted): Add experimental page

### PR DESCRIPTION
**Pull request by hubertdeng123**
## DESCRIBE YOUR PR
*This moves some self-hosted pages to an experimental section to boldly indicate that these features are not well supported

## IS YOUR CHANGE URGENT?  

Help us prioritize incoming PRs by letting us know when the change needs to go live.
- [ ] Urgent deadline (GA date, etc.): 
- [ ] Other deadline: 
- [x] None: Not urgent, can wait up to 1 week+

## SLA

- Teamwork makes the dream work, so please add a reviewer to your PRs.
- Please give the docs team up to 1 week to review your PR unless you've added an urgent due date to it.
Thanks in advance for your help!

## PRE-MERGE CHECKLIST

*Make sure you've checked the following before merging your changes:*

- [ ] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs (subject matter experts)
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## LEGAL BOILERPLATE

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

## EXTRA RESOURCES

- [Sentry Docs contributor guide](https://docs.sentry.io/contributing/)

### CODE REVIEW

File: develop-docs/self-hosted/experimental/index.mdx [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+title: Experimental Configurations
+---
+
+These pages here are purely experimental and are not officially supported workflows. Much of this is contributed to by our wonderful developer community and we hope it is useful for those customizing self-hosted Sentry to your specific needs. Please use at your own risk.
```
Comment (azaslavsky) [inline develop-docs/self-hosted/experimental/index.mdx:5]:
```suggestion
These pages here are purely experimental and are not officially supported workflows. Much of this is contributed by our wonderful developer community. We hope it is useful for those customizing self-hosted Sentry to your specific needs, but are unable to offer official support or triage for issues arising from these workflows. **Use at your own risk!**
```

**Review by azaslavsky — APPROVED** [review]

File: develop-docs/self-hosted/experimental/reverse-proxy.mdx [hunk 2]
```diff
@@ -4,6 +4,10 @@ sidebar_title: Reverse Proxy
 sidebar_order: 80
 ---
 
+<Alert title="Important" level="warning">
+    The docs below are contributed by the community. Sentry does not officially provide support for setting up reverse proxies for your self-hosted Sentry setup. Follow these guidelines at your own risk as these docs are not maintained nor kept up to date!
```
Comment (joshuarli) [inline develop-docs/self-hosted/experimental/reverse-proxy.mdx:8]: I think it'd be better to keep this banner generic so we can copy paste this to every file under this `experimental` section. (Can you do that as well?)

```suggestion
    These are community-contributed docs. Sentry does not officially provide support for self-hosted configurations beyond the defaults.
```

**Review by joshuarli — CHANGES_REQUESTED** [review]
I think we should also move Geolocation under this section, it's not a core feature, and it's not trivial to set up (and maintain) correctly.

**Comment by hubertdeng123** [comment]
> I think we should also move Geolocation under this section, it's not a core feature, and it's not trivial to set up (and maintain) correctly.

This ships with self-hosted, so I don't think it should belong here. Good call on CSP

File: develop-docs/self-hosted/experimental/reverse-proxy.mdx (hunk 2 above)
Comment (hubertdeng123) [reply develop-docs/self-hosted/experimental/reverse-proxy.mdx:8]: CSP docs are written by us but are technically experimental as well so I don't think we should necessarily have generic banners for every file under the `experimental` section. I'm on board with the same generic banner for community contributed docs though

**Review by joshuarli — APPROVED** [review]

**Comment by aldy505** [comment]
Since you made this, can I put some more obscure stuff inside?

**Comment by hubertdeng123** [comment]
@aldy505 depends on what kind of stuff, what are you thinking?

**Pull request merged**